zkevm-circuits/src/exp_circuit.rs

```diff
@@ -382,37 +401,137 @@ impl<F: Field> ExpCircuitConfig<F> {
 
         Ok(())
     }
+    /// Assign witness to the exponentiation circuit.
+    pub fn assign_exp_events(
```

Comment:
you should move the `pub fn` above all the private functions, simply to follow idiomatic rust practices